__tests__/CollapsedMetrics.test.js

```diff
@@ -0,0 +1,851 @@
+    });
+});
+
+describe('getFirstRepAvgVelocity', () => {
+    
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+            
+        let result = sut.getFirstRepAvgVelocity(set);
+            
+        expect(result).toBe(null);        
+    });
+    
+    test('returns first rep avg velocity', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: false,
+                    removed: false,
+                    data: [-3456, 40, 3.15]
+                },
+                {
+                    isValid: true,
+                    removed: true,
+                    data: [-3456, 40, 1.15]
+                },
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                }, 
+                {
+                    isValid: false,
+                    removed: true,
+                    data: [-3456, 40, 2.15]
+                },
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
+                },
+            ]
+        }
+            
+        let result = sut.getFirstRepAvgVelocity(set);
+    
+        expect(result).toBe(1.83);
+    });
+});
+    
+describe('getLastRepAvgVelocity', () => {
+    
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+            
+        let result = sut.getLastRepAvgVelocity(set);
+            
+        expect(result).toBe(null);        
+    });
+    
+    test('returns last rep avg velocity', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: false,
+                    removed: false,
+                    data: [-3456, 40, 3.15]
+                },
+                {
+                    isValid: true,
+                    removed: true,
+                    data: [-3456, 40, 1.15]
+                },
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                },
+                {
+                    isValid: false,
+                    removed: true,
+                    data: [-3456, 40, 2.15]
+                },
+                {
+                    isValid: false,
+                    removed: true,
+                    data: [-3456, 40, 1.00]
+                },  
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
```

Comment:
- [ ] Make some of the invalid and removed things need to be the last 2 in the list to properly test last rep